src/agreement/mod.rs

```diff
@@ -481,7 +482,7 @@ impl<NodeUid: Clone + Debug + Ord> Agreement<NodeUid> {
         sender_id: &NodeUid,
         msg: CommonCoinMessage,
     ) -> Result<FaultLog<NodeUid>> {
-        let coin_step = self.common_coin.handle_message(sender_id, msg)?;
+        let coin_step = self.common_coin.handle_message(sender_id, msg).map_err(Error::HandleCoinCommonCoin)?;
```

Comment:
`.map_err(Error::HandleCoinCommonCoin)`
could be
`.map_err(|e| Error::CommonCoin(e, line!()))`.

Reply:
The inclusion of the line information is redundant though. The error variants themselves include all of the information needed to trace the precise point of origin both programatically and via display/debug messages. Also, as Mark alluded to, including this kind of information comes at a cost.